x-pack/apm-server/aggregation/lsm.go

```diff
@@ -108,17 +113,17 @@ func wrapNextProcessor(processor modelpb.BatchProcessor) aggregators.Processor {
 	}
 }
 
-func removeRUMGlobalLabels(event *modelpb.APMEvent) {
-	// Remove global labels for RUM services to avoid explosion of metric groups
-	// to track for servicetxmetrics.
-	// For consistency, this will remove labels for other aggregated metrics as well.
-	switch event.GetAgent().GetName() {
+// Check Elastic Agent belongs to rum-services
+func isRumAgent(name string) bool {
+	switch name {
 	case "rum-js", "js-base", "android/java", "iOS/swift":
-	default:
-		return
+		return true
 	}
+	return false
+}
 
-	// Setting the labels to non-global so that they are ignored by the aggregator.
+// Setting the labels to non-global so that they are ignored by the aggregator.
+func removeGlobalLabels(event *modelpb.APMEvent) {
```

Comment:
The structure you are proposing reads nicely. However, we are not intending, and it would be harmful, to remove global labels for anything non-rum in the future. Therefore I prefer the current code which is encapsulating the logic into one combined function. It prevents our future selves from just reusing the `removeGlobalLabels` function without the `isRumAgent` safeguard. 

Having this said, both implementations generally look fine to me, but I don't see an improvement by your suggestion over what is currently implemented.